common/src/main/java/com/wynntils/commands/QuestCommand.java

```diff
@@ -0,0 +1,271 @@
+                .then(Commands.literal("wiki")
+                        .then(Commands.argument("quest", StringArgumentType.greedyString())
+                                .suggests(QUEST_SUGGESTION_PROVIDER)
+                                .executes(this::lookupOnWiki)))
+                .executes(this::syntaxError);
+    }
+
+    private int listQuests(CommandContext<CommandSourceStack> context, String sort) {
+        QuestSortOrder order = QuestSortOrder.fromString(sort);
+
+        Managers.Quest.rescanQuestBook(true, false);
+
+        if (Managers.Quest.getQuestsRaw().isEmpty()) {
+            context.getSource()
+                    .sendSuccess(
+                            Component.literal("Quest Book was not scanned. You might have to retry this command.")
+                                    .withStyle(ChatFormatting.YELLOW),
+                            false);
+        }
+
+        List<QuestInfo> quests = Managers.Quest.getQuests(order).stream()
+                .filter(QuestInfo::isTrackable)
+                .toList();
+
+        if (quests.isEmpty()) {
+            context.getSource()
+                    .sendFailure(Component.literal("No active quests found!").withStyle(ChatFormatting.RED));
+            return 1;
+        }
+
+        MutableComponent response = Component.literal("Active quests:").withStyle(ChatFormatting.AQUA);
+        generateQuestList(quests, response);
+
+        context.getSource().sendSuccess(response, false);
+        return 1;
+    }
+
+    private int searchQuests(CommandContext<CommandSourceStack> context) {
+        String searchText = context.getArgument("text", String.class);
+
+        Managers.Quest.rescanQuestBook(true, false);
+
+        if (Managers.Quest.getQuestsRaw().isEmpty()) {
+            context.getSource()
+                    .sendSuccess(
+                            Component.literal("Quest Book was not scanned. You might have to retry this command.")
+                                    .withStyle(ChatFormatting.YELLOW),
+                            false);
+        }
+
+        List<QuestInfo> quests;
+        MutableComponent response;
+
+        quests = Managers.Quest.getQuestsRaw().stream()
+                .filter(quest -> StringUtils.initialMatch(quest.getName(), searchText)
+                        || StringUtils.initialMatch(quest.getNextTask(), searchText))
+                .toList();
+
+        if (quests.isEmpty()) {
+            context.getSource()
+                    .sendFailure(Component.literal("No matching quests found!").withStyle(ChatFormatting.RED));
+            return 1;
+        }
+
+        response = Component.literal("Matching quests:").withStyle(ChatFormatting.AQUA);
+
+        generateQuestList(quests, response);
+
+        context.getSource().sendSuccess(response, false);
+        return 1;
+    }
+
+    private void generateQuestList(List<QuestInfo> quests, MutableComponent response) {
```

Comment:
Can you sort methods in some order? I usually do command methods first, then helper methods.